core/src/test/java/org/infinispan/conflict/impl/BaseMergePolicyTest.java

```diff
@@ -144,45 +143,24 @@ public void testPartitionMergePolicy() {
    }
 
    protected <K, V> AdvancedCache<K, V> getCacheFromPreferredPartition() {
-      AdvancedCache[] caches = caches().stream().map(AdvancedCache.class::cast).toArray(AdvancedCache[]::new);
+      AdvancedCache[] caches = caches().stream().map(Cache::getAdvancedCache).toArray(AdvancedCache[]::new);
       return getCacheFromPreferredPartition(caches);
    }
 
    protected <K, V> AdvancedCache<K, V> getCacheFromPreferredPartition(AdvancedCache... caches) {
-      // Emulate the algorithm in PreferAvailabilityStrategy/PreferConsistencyStrategy to determine the
-      // preferred topology and pick the cache on the node that will send it during cluster status recovery.
-      List<KeyValuePair<AdvancedCache, CacheStatusResponse>> statusResponses =
-         Arrays.stream(caches).map(c -> new KeyValuePair<>(c, getCacheStatus(c)))
-               .sorted(Comparator.comparing(KeyValuePair::getValue, PreferAvailabilityStrategy.RESPONSE_COMPARATOR))
-               .collect(Collectors.toList());
-
-      CacheTopology maxStableTopology = null;
-      for (KeyValuePair<AdvancedCache, CacheStatusResponse> response : statusResponses) {
-         CacheTopology stableTopology = response.getValue().getStableTopology();
-         if (stableTopology == null) continue;
-
-         if (maxStableTopology == null || maxStableTopology.getMembers().size() < stableTopology.getMembers().size()) {
-            maxStableTopology = stableTopology;
-         }
-      }
-
-      AdvancedCache<K, V> maxTopologySender = null;
-      CacheTopology maxTopology = null;
-
-      for (KeyValuePair<AdvancedCache, CacheStatusResponse> response : statusResponses) {
-         CacheTopology stableTopology = response.getValue().getStableTopology();
-         if (!Objects.equals(stableTopology, maxStableTopology)) continue;
-
-         CacheTopology topology = response.getValue().getCacheTopology();
-         if (topology == null) continue;
-
-         if (maxTopology == null || maxTopology.getMembers().size() < topology.getMembers().size()) {
-            maxTopology = topology;
-            maxTopologySender = response.getKey();
-         }
-      }
-      if (trace) log.tracef("getCacheFromPreferredPartition: partition=%s", maxTopology != null ? maxTopology.getCurrentCH().getMembers() : null);
-      return maxTopologySender;
+      Map<Address, CacheStatusResponse> statusResponses =
+         Arrays.stream(caches).map(c -> new KeyValuePair<>(address(c), getCacheStatus(c)))
+               .collect(Collectors.toMap(KeyValuePair::getKey, KeyValuePair::getValue));
```

Comment:
Can be just: `Arrays.stream(caches).collect(Collectors.toMap(this::address, this::getCacheStatus));`

Reply:
Here too, I needed a `LinkedHashMap` at some point but it's too much now